### PR TITLE
ios: optimise chat switching

### PR DESCRIPTION
**Pull request by levitatingpineapple**
- move `apiGetChat` to background thread
- remove redundant `.onChange(of: chatModel.chatId)` modifier
- during load - hide items instead of removing them
- animate item appearance

### CODE REVIEW

File: apps/ios/Shared/Model/SimpleXAPI.swift [hunk 1]
```diff
@@ -333,23 +333,29 @@ func apiGetChatItems(type: ChatType, id: Int64, pagination: ChatPagination, sear
 }
 
 func loadChat(chat: Chat, search: String = "") {
-    do {
-        let cInfo = chat.chatInfo
-        let m = ChatModel.shared
-        let im = ItemsModel.shared
-        m.chatItemStatuses = [:]
-        im.reversedChatItems = []
```
Comment (levitatingpineapple) [inline apps/ios/Shared/Model/SimpleXAPI.swift:341]: Removal of `im.reversedChatItems = []`
Chat items are now hidden during load

File: apps/ios/Shared/Views/Chat/ChatView.swift [hunk 2]
```diff
@@ -221,75 +221,79 @@ struct ChatView: View {
...
-                if selectedChatItems != nil {
-                    Button {
-                        withAnimation {
-                            selectedChatItems = nil
-                        }
-                    } label: {
-                        Text("Cancel")
-                    }
+                if im.isLoading {
+                    ProgressView().modifier(AppearAfter(seconds: 0.5))
```
Comment (levitatingpineapple) [inline apps/ios/Shared/Views/Chat/ChatView.swift:225]: Large diff due to indentation - only added `ProgressView()`

Comment (epoberezkin) [reply apps/ios/Shared/Views/Chat/ChatView.swift:225]: indentation is not diff, GitHub hides it

File: apps/ios/Shared/Views/Chat/Group/GroupMemberInfoView.swift [hunk 3]
```diff
@@ -321,24 +321,24 @@ struct GroupMemberInfoView: View {
...
     }
 
     func newDirectChatButton(_ contactId: Int64, width: CGFloat) -> some View {
         InfoViewButton(image: "message.fill", title: "message", width: width) {
-            do {
-                let chat = try apiGetChat(type: .direct, id: contactId)
-                chatModel.addChat(chat)
-                dismissAllSheets(animated: true)
-                DispatchQueue.main.async {
-                    chatModel.chatId = chat.id
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/Group/GroupMemberInfoView.swift:338]: what did it do?

File: apps/ios/Shared/Views/Chat/ReverseList.swift [hunk 4]
```diff
@@ -32,7 +32,7 @@ struct ReverseList<Item: Identifiable & Hashable & Sendable, Content: View>: UIV
             case let .item(id):
                 controller.scroll(to: items.firstIndex(where: { $0.id == id }), position: .bottom)
             case .bottom:
-                controller.scroll(to: 0, position: .top)
+                controller.scroll(to: nil, position: .top)
```
Comment (epoberezkin) [inline apps/ios/Shared/Views/Chat/ReverseList.swift:35]: nil?

**Pull request merged**